modules/backup-s3/client.go

```diff
@@ -33,9 +34,13 @@ type s3Client struct {
 	config   *clientConfig
 	logger   logrus.FieldLogger
 	dataPath string
+	bucket   string
+	path     string
+	creds    *credentials.Credentials
```

Comment:
when you would refactor the way that you construct the client as suggested [here](https://github.com/weaviate/weaviate/pull/6012/files#r1814432476) then `bucket` `path` and `creds` are not needed variables and could be removed, which would further simplify the whole implementation

Reply:
done